src/domain/services/AIService.ts

```diff
@@ -371,16 +369,29 @@ export class AIService {
       );
     }
 
-    const ais = await prismadb.aI.findMany({
-      select: {
-        ...listAIResponseSelect(),
-        chats: {
-          where: {
-            userId,
-            isDeleted: false,
-          },
+    let categoryType : CategoryType | undefined;
+    try {
+      // @ts-ignore
+      categoryType = CategoryTypesHardcoded[categoryId] as CategoryType
+    } catch (error) {
+      console.error("Error parsing category type", error);
+      categoryType = undefined;
+    }
+
+    let selectClause = {...listAIResponseSelect(), chats: {
+        where: {
+          userId,
+          isDeleted: false,
         },
       },
+    }
+
+    if (categoryType != undefined && categoryId != "NONE") {
+      selectClause["aicategorytypes"] = {where: {categoryType: categoryType}};
+    }
```

Comment:
This doesn't look like for filtering by category. We need a clause similar to [getGroupCriteria](https://github.com/nicolasdesmarais/ai-companion/blob/main/src/domain/services/AIService.ts#L757-L768). 

You can keep the condition above: 
```
if (categoryId) {
  whereCondition.AND.push(this.getCategoryCriteria(categoryId));
}
```

```
  private getCategoryCriteria(categoryId: string) {
    return {
        aicategorytypes: {
          some: {
            categoryType: categoryId,
          },
        },
      }
  }
```

Reply:
Implemented